Source/Extensions/Blazorise.LoadingIndicator/LoadingIndicator.razor.cs

```diff
@@ -0,0 +1,312 @@
+                }
+            }
+
+            DirtyClasses();
+            DirtyStyles();
+
+            return base.SetParametersAsync( parameters );
+        }
+
+        protected override void Dispose( bool disposing )
+        {
+            if ( disposing )
+            {
+                Service = null;
+            }
+
+            base.Dispose( disposing );
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Workaround for issue https://github.com/dotnet/aspnetcore/issues/15311
+        /// Setting svg width or height to null if it had a value before throws an exception
+        /// https://icons8.com/preloaders/en/search/spinner#
+        /// </summary>
+        private string SpinnerSVG =>
+            @$"<svg viewBox='0 0 128 128' 
+                    {( !string.IsNullOrEmpty( SpinnerWidth ) ? $"width='{SpinnerWidth}'" : "" )}
+                    {( !string.IsNullOrEmpty( SpinnerHeight ) ? $"height='{SpinnerHeight}'" : "" )}>
+                      <g>
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerColor.Name}' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(45 64 64)' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(90 64 64)' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(135 64 64)' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(180 64 64)' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(225 64 64)' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(270 64 64)' />
+                          <path d = 'M38.52 33.37L21.36 16.2A63.6 63.6 0 0 1 59.5.16v24.3a39.5 39.5 0 0 0-20.98 8.92z' fill='{SpinnerBackground.Name}' transform='rotate(315 64 64)' />
+                          <animateTransform attributeName = 'transform' type='rotate' values='0 64 64;45 64 64;90 64 64;135 64 64;180 64 64;225 64 64;270 64 64;315 64 64' calcMode='discrete' dur='720ms' repeatCount='indefinite' />
+                      </g>
+                  </svg>";
+
+
+        /// <summary>
+        /// Service used to control this instance.
+        /// </summary>
+        [Parameter]
+        public LoadingIndicatorService Service
```

Comment:
Can service be injected by default without for need to manually set it through parameter?

Reply:
Yes, having to set Service feels non-standard somehow but it makes the component more generic. If a service is injected by default then every indicator will get an instance of the same service and that's not good for a scenario where you want to use loading indicator for some form on the page for example. I think an **injected** service is really only for the global app only scenario, so you can define the wrapper in the main layout but control it from elsewhere. Maybe some hybrid approach? have a boolean param Default like <LoadingIndicator Default /> then it will use the service from DI. Some syntax sugar. Let me know what you think.

Reply:
I think by default, we can have it injected. And then, with a parameter, we can override it.

Reply:
Ok, I trust your judgement here. But I feel it will be ackward in scenario like 

```
<LoadingIndicator @ref="textBusy" **Service="null"**>
   <TextEdit/>
</LoadingIndicator>
```

Reply:
I removed Service property. Service is now limited to global application busy scenario. So whoever is using this Extension component will get a wrapper to use with other components and an out of the box global app wrapper with service, the multiple-indicators scenario is something the devs will have to do on their own, by inheriting from LoadingIndicator and adding Service parameter for example.

```
// Config.cs
services.AddLoadingIndicator(); // adds scoped ILoadingIndicator service

// MainLayout.cs
@Body
<ApplicationLoadingIndicator /> <!-- no need to wrap anything since it's fullscreen -->

// Page.cs
@inject LoadingIndicatorService AppBusy

<Button Clicked="() => AppBusy.Show()" />
```
https://user-images.githubusercontent.com/22550674/180569800-f6a05c54-1087-491e-8ea6-9c6acfdc443d.mp4